### ISSUE TITLE
New syntax error seems inaccurate

### ISSUE THREAD

**Issue by davidtheclark** [postcss/postcss#521]
When I upgraded to 5.0.2 I got the following syntax error, which seems inaccurate:

```
/Users/davidclark/dev/stylelint/node_modules/postcss/lib/lazy-result.js:172
        if (this.error) throw this.error;
                                  ^
CssSyntaxError: <css input>:1:46: Unclosed quote
a::before { background: url( 'asdf(Vcxvsd)ASD' ); }
                                             ^
    at Input.error (/Users/davidclark/dev/stylelint/node_modules/postcss/lib/input.js:61:21)
    at unclosed (/Users/davidclark/dev/stylelint/node_modules/postcss/lib/tokenize.js:50:21)
    at Object.tokenize [as default] (/Users/davidclark/dev/stylelint/node_modules/postcss/lib/tokenize.js:140:38)
    at Parser.tokenize (/Users/davidclark/dev/stylelint/node_modules/postcss/lib/parser.js:49:44)
    at parse (/Users/davidclark/dev/stylelint/node_modules/postcss/lib/parse.js:24:12)
    at new LazyResult (/Users/davidclark/dev/stylelint/node_modules/postcss/lib/lazy-result.js:57:24)
    at Processor.process (/Users/davidclark/dev/stylelint/node_modules/postcss/lib/processor.js:36:16)
```

Does that seem like a bug to you?

**Comment by ai** [comment]
Yeah, I change `url()` parsing. PostCSS 5.0 parses `url()` as separated token, not as common function (this is true by CSS spec).

And this new parser has some optimization. It thinks that `(` and quote will not be splited with space.

I can add space support, but it will slowdown parser and make code much complicated.

So can you show me real world example?

**Comment by davidtheclark** [comment]
Hm, ok. The real world example is just if people want to write their functions with spaces inside parentheses and with quotes for `url()`, right?

```css
background-image: url( 'somepic.png' );
```

Looks to me like the spec definitely does allow for whitespace within the parens: `url\({w}{string}{w}\)`.

My own take would to be to make sure that PostCSS can handle any valid CSS syntax. However, if you think this is rare enough that PostCSS is not even going to support it, I'll just have to change stylelint a little so that stylelint doesn't support it either.

**Comment by ben-eb** [comment]
> My own take would to be to make sure that PostCSS can handle any valid CSS syntax.

:+1:

**Comment by ai** [comment]
@ben-eb @davidtheclark Are you know that by CSS spec any CSS symbol can be replaed by it code `\0013`. So we can replace `{` by `\dddd` and it still be valid.

Of course, PostCSS doesn’t parse it.

**Comment by ai** [comment]
I try to find best way to make parser smarter, but I have good solution right now.

**Comment by davidtheclark** [comment]
@ai Didn't know — and certainly trust that you've thought about this stuff more than I have :)

**Comment by lydell** [comment]
@ai That is not true. You can _not_ replace replace any symbol with an escape and still have an equivalent style sheet. Try to replace the opening `{` of a block with an escape—the block rule will no longer work (that can also mean that no browser follows the spec either, but I’m certain that’s not the case). That’s the point of escapes: escaping the regular meaning of a character. This allows you to use for example `{` as part of a class name (which is terribly useless, but valid).

Check out these [railroad diagrams from the spec](http://www.w3.org/TR/css-syntax-3/#token-diagrams) if you don’t believe me. Also see this [handy CSS escaping tool](https://mothereff.in/css-escapes).

In the end, though, I’ve never used an escape in CSS (other than in strings, such as `a::after{content:'\A';white-space:pre;}` to render a newline after some element), and I’ve never seen anyone else doing it either, so I agree that it doesn’t matter much practically. But the error message @davidtheclark got is not very good.

**Comment by ai** [comment]
I think I found good way to handle it: 6567bfc

**Comment by ai** [comment]
Released in 5.0.3.
